src/pkg/packaging-tools/sharedFramework/sharedFramework.csproj

```diff
@@ -113,18 +122,69 @@
       Overwrite="true" />
   </Target>
 
+  <!--
+    NETCoreApp (and only NETCoreApp) carries the host and host resolver in the shared framework,
+    which lives outside the shared/ directory. These were included in publish through package
+    references and the files must be moved.
+  -->
+  <Target Name="MoveSharedFrameworkHost"
+          Condition="'$(SfxIdentity)' == 'Microsoft.NETCore.App'"
+          AfterTargets="Publish">
+    <!-- Use the muxer we intended to consume (either restored a shipped version or the one we built) -->
+    <Move SourceFiles="$(PublishDir)dotnet$(ExeSuffix)" DestinationFolder="$(SharedFrameworkLayoutDir)" />
+
+    <!--  Use the FXR Resolver we intended to consume (either restored a shipped version or the one we built) -->
+    <Move SourceFiles="$(PublishDir)$(LibPrefix)hostfxr$(LibSuffix)" DestinationFolder="$(SharedFrameworkLayoutDir)\host\fxr\$(HostResolverVersion)" />
+
+    <!-- Ensure the host is executable. See https://github.com/NuGet/Home/issues/4424 -->
+    <Exec Command="chmod u+x $(SharedFrameworkLayoutDir)dotnet$(ExeSuffix)"
+          Condition="'$(OSGroup)' != 'Windows_NT'" />
+  </Target>
+
   <!-- Clean up artifacts that dotnet-publish generates which we don't need -->
   <Target Name="DeleteUnnecessaryPublishOutputs" AfterTargets="Publish">
     <ItemGroup>
       <ToDelete Include="$(PublishDir)apphost$(ExeSuffix)" />
       <ToDelete Include="$(PublishDir)$(LibPrefix)comhost$(LibSuffix)" />
       <ToDelete Include="$(PublishDir)$(LibPrefix)hostfxr$(LibSuffix)" />
       <ToDelete Include="$(PublishDir)$(LibPrefix)ijwhost$(LibSuffix)" />
+      <ToDelete Include="$(PublishDir)$(LibPrefix)ijwhost.lib" />
```

Comment:
Can you explain the difference between these?

```
      <ToDelete Include="$(PublishDir)$(LibPrefix)ijwhost$(LibSuffix)" />
      <ToDelete Include="$(PublishDir)$(LibPrefix)ijwhost.lib" />
```

Reply:
This is because `LibSuffix` only handles dynamic libraries (`.dll`/`.so`), but we also build ijwhost as a static library (`.lib`/`.a`/?). It's only built on Windows so I don't have to worry about handling other extensions for now.

Adding a comment to explain.